static/src/javascripts/projects/common/modules/navigation/sticky.js

```diff
@@ -0,0 +1,287 @@
+            });
+        });
+    };
+
+    StickyHeader.prototype.setScrollDirection = function (scrollY) {
+        this.config.direction = (scrollY > this.config.prevScroll) ? 'down' : 'up';
+        this.config.prevScroll = scrollY;
+    };
+
+    StickyHeader.prototype.shouldShowNavigation = function (scrollY) {
+        if (this.config.direction === 'up' && this.config.distance === 0) {
+            this.config.distance = scrollY;
+        }
+
+        // If distance scrolled is more than showNavigationDepth show navigation
+        this.config.showNavigation = (Math.abs(scrollY - this.config.distance) > this.config.showNavigationDepth);
+    };
+
+    StickyHeader.prototype.showNavigation = function (scrollY) {
+        this.shouldShowNavigation(scrollY);
+
+        // If user is scrolling up and navigation threshold was met show navigation
+        if (this.config.direction === 'up' && this.config.showNavigation) {
+            fastdom.write(function () {
+                if (this.isTablet || this.isMobile) {
+                    this.$els.navigation.removeClass('animate-down-mobile').addClass('animate-up-mobile');
+                } else {
+                    this.$els.navigation.removeClass('animate-down-desktop').addClass('animate-up-desktop');
+                }
+            }.bind(this));
+        } else {
+            // If user is scrolling down and navigation is visible reset bounce distance
+            if (this.config.showNavigation) {
+                // Reset distance bouncing
+                this.config.distance = 0;
+
+                // Close meganav if it's open
+                if (this.$els.navHeader.hasClass('navigation-container--expanded')) {
+                    bean.fire(qwery('.js-navigation-toggle')[0], 'click');
+                }
+            }
+            fastdom.write(function () {
+                if (this.isTablet || this.isMobile) {
+                    this.$els.navigation.removeClass('animate-up-mobile').addClass('animate-down-mobile');
+                } else {
+                    this.$els.navigation.removeClass('animate-up-desktop').addClass('animate-down-desktop');
+                }
+            }.bind(this));
+        }
+    };
+
+    StickyHeader.prototype.setNavigationDefault = function () {
+        fastdom.write(function () {
+            // Make sure navigation is visible and has its default styles
+            this.$els.navigation.removeAttr('class');
+            this.$els.navigation.attr('class', this.navigationClassList);
+        }.bind(this));
+    };
+
+    StickyHeader.prototype.updatePosition = function (breakpoint) {
+        var bannerHeight = this.$els.bannerDesktop.dim().height || 128,
+            scrollY      = this.$els.window.scrollTop();
+
+        this.setScrollDirection(scrollY);
+
+        // Header is slim and navigation is shown on the scroll up
+        if (scrollY >= this.headerBigHeight + (bannerHeight * this.config.showHeaderDepth)) {
```

Comment:
might be worth making the contents of these conditional blocks separate functions too, so you can read the logic and functionality separately?